cmd/bridge/main.go

```diff
@@ -240,6 +241,16 @@ func main() {
 		}
 	}
 
+	nodeArchitectures := strings.Split(*fNodeArchitectures, ",")
+	if *fNodeArchitectures != "" {
+		for _, str := range nodeArchitectures {
+			if str == "" {
+				bridge.FlagFatalf("node-architectures", "list must contain name of node architectures separated by comma")
+			}
+		}
+	}
+	nodeArchitectures = serverutils.RemoveEmptyStrings(nodeArchitectures)
```

Comment:
`bridge.FlagFatalf` exits the application if an empty string is found in `nodeArchitectures`, so this line shouldn't be necessary.

Reply:
Right, but there is a know behaviour of the `strings.Split()` method where
`strings.Split("", ",")`
will return
`[“”]`

thats why we are removing them extra.

Reply:
So the results of string.Split are stored in `nodeArchitectures` on L244, then you loop through all of the elements in `nodeArchitectures` on L246-250 and exit if any empty string is found. I still don't understand why you would then need to have logic to remove empty strings from that same array. The previous validation logic would have already exited with an error message if any empty string existed in `nodeArchitectures`.

Reply:
I see your point and to be honest I dont thing there is a clean way how to solve it due to the issue mentioned above... then Im debating if it wont be better 
```go
	nodeArchitectures := []string{}
	if *fNodeArchitectures != "" {
		for _, str := range strings.Split(*fNodeArchitectures, ",") {
			str = strings.TrimSpace(str)
			if str == "" {
				bridge.FlagFatalf("node-architectures", "list must contain name of node architectures separated by comma")
			}
			nodeArchitectures = append(nodeArchitectures, str)
		}
	}
```